registry/json.go

```diff
@@ -21,3 +28,41 @@ func (registry *Registry) getJson(url string, response interface{}) error {
 
 	return nil
 }
+
+// getPaginatedJson accepts a string and a pointer, and returns the
+// next page URL while updating pointed-to variable with a parsed JSON
+// value. When there are no more pages it returns `ErrNoMorePages`.
+func (registry *Registry) getPaginatedJson(url string, response interface{}) (string, error) {
+	resp, err := registry.Client.Get(url)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil {
+		return "", err
+	}
+
+	decoder := json.NewDecoder(resp.Body)
+	err = decoder.Decode(response)
+	if err != nil {
+		return "", err
+	}
+	return getNextLink(resp)
+}
+
+var linkRE *regexp.Regexp = regexp.MustCompile(`^ *<?([^;>]+)>?(?:; *([a-z]+)="?([^";]*)"?)*$`)
```

Comment:
> Looking around, you're also matching both : and ; as a separator between the url and the link.

I don't think so -- maybe you're scanning `(?:;` as a capturing group starting with a colon, rather than a non-capturing group starting with a semi-colon? Regexps, eh.